lib/elastomer/client.rb

```diff
@@ -228,6 +238,20 @@ def request( method, path, params )
     end
     # rubocop:enable Metrics/MethodLength
 
+    # Internal: Returns a new Elastomer::Client error that wraps the given
+    # Faraday error. A generic Error is returned if we cannot wrap the given
+    # Faraday error.
+    #
+    #   error  - The Faraday error
+    #   method - The request method
+    #   path   - The request path
+    #
+    def wrap_faraday_error(error, method, path)
+      error_name  = error.class.name.split("::").last
+      error_class = Elastomer::Client.const_get(error_name) rescue Elastomer::Client::Error
```

Comment:
Does this get a specific class matching the server's Java exception?

Reply:
I think it's mapping back to one of [these](https://github.com/github/elastomer-client/blob/master/lib/elastomer/client/errors.rb#L76-L84)